boot/zephyr/serial_adapter.c

```diff
@@ -147,8 +147,12 @@ boot_uart_fifo_callback(struct device *dev)
 		if (cur < CONFIG_BOOT_MAX_LINE_INPUT_LEN) {
 			cmd->line[cur++] = byte;
 		}
-
-		if (byte ==  '\n') {
+		#ifdef CONFIG_BOOT_SERIAL_UART_DETECT
+		if(byte == '\r'){
```

Comment:
Spacing. Also, do we want to treat '\r' as if it were a '\n', or just ignore it. Most terminals should be sending either just a LF, or a CR/LF pair.

Reply:
In nrf52832 serial_adapter read uart always '\r', No matter how I set up the serial tool, I can't get '\n', Maybe It is a issue in uart drv of nrf52832. If you have other hw platform, Could you help double confirm it? Thanks

Reply:
I suppose it doesn't hurt anything to allow the '\r' as well, but it does make me wonder how the code was used by others.

Reply:
In boot_uart_fifo_callback have below code:
```
if (byte ==  '\n'){
	cmd->len = cur;
	sys_slist_append(&lines_queue, &cmd->node);
	cur = 0;
	cmd = NULL;
}
```
It means only meeting '\n' , add cmd to list.  
I hope press **Enter** key to goto recovery, During mcuboot startup,long press Enter key,  boot_uart_fifo_callback   always gets '\r' from serial , It casue no new line to list,  main can't read newline by console_read cause can't go into recovery mode.
So I convert '\r' to '\n' to get newline and go into recovery mode

Reply:
To be analyzed...